core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -397,9 +398,35 @@ abstract class AppBase(
   }
 
   protected def postCompletion(): Unit = {
+    updateAttemptIdTracker()
```

Comment:
That's too late. If the eventlog is large and it is taking time for a specific thread to complete its eventlog (due to GC for example), then other threads will be done before the current thread catches them
Instead, each thread should register its AppId-AttemptId during handling of the `SparkListenerApplicationStart Handler.`
P.S: I saw in some of the customer eventlogs, 2 environmentUpdate events in the log files. So, we should be careful with this corner case.